src/agents/memory/providers/redis.py

```diff
@@ -0,0 +1,366 @@
+        """
+        client = await self._get_redis_client()
+        session_data: dict[str, str] = await client.hgetall(self.session_key)  # type: ignore[misc]
+
+        return session_data if session_data else None
+
+    async def get_session_size(self) -> int:
+        """Get the number of messages in the session.
+
+        Returns:
+            Number of messages in the session
+        """
+        client = await self._get_redis_client()
+        length: int = await client.llen(self.messages_key)  # type: ignore[misc]
+        return length
+
+    async def close(self) -> None:
+        """Close the Redis connection."""
+        if self._redis_client is not None:
+            await self._redis_client.aclose()
+            self._redis_client = None
+
+    async def __aenter__(self):
+        """Async context manager entry."""
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb):
+        """Async context manager exit."""
+        await self.close()
+
+
+class RedisSessionManager:
+    """Manager class for Redis sessions with connection pooling."""
+
+    def __init__(
+        self,
+        redis_url: str = "redis://localhost:6379",
+        db: int = 0,
+        session_prefix: str = "agent_session",
+        messages_prefix: str = "agent_messages",
+        default_ttl: int | None = None,
+        max_connections: int = 10,
+    ):
+        """Initialize the Redis session manager.
+
+        Args:
+            redis_url: Redis connection URL
+            db: Redis database number
+            session_prefix: Prefix for session metadata keys
+            messages_prefix: Prefix for message list keys
+            default_ttl: Default TTL for sessions in seconds
+            max_connections: Maximum number of Redis connections in the pool
+        """
+        self.redis_url = redis_url
+        self.db = db
+        self.session_prefix = session_prefix
+        self.messages_prefix = messages_prefix
+        self.default_ttl = default_ttl
+
+        self._redis_pool = redis.ConnectionPool.from_url(
+            redis_url,
+            db=db,
+            max_connections=max_connections,
+            decode_responses=True,
+        )
+
+    def get_session(
+        self,
+        session_id: str,
+        ttl: int | None = None,
+    ) -> RedisSession:
+        """Get a Redis session instance.
+
+        Args:
+            session_id: Unique identifier for the session
+            ttl: TTL for this session. If None, uses default_ttl
+
+        Returns:
+            RedisSession instance
+        """
+        session = RedisSession(
+            session_id=session_id,
+            redis_url=self.redis_url,
+            db=self.db,
+            session_prefix=self.session_prefix,
+            messages_prefix=self.messages_prefix,
+            ttl=ttl or self.default_ttl,
+        )
+
+        # Share the connection pool
+        session._redis_client = redis.Redis(connection_pool=self._redis_pool)
```

Comment:
instead of accessing a private property this way, simply enabling to pass a shared Redis instance should be straight-forward. So, this means users can pass 1) redis_url or 2) redis_client.

With this way, this RedisSessionManager class is not necessary as part of this SDK. If you want to have something like this, please have the code on your app side.